staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_external.go

```diff
@@ -1060,6 +1060,16 @@ func ensureStaticIP(s CloudAddressService, name, serviceName, region, existingIP
 		existed = true
 	}
 
+	// If address exists, get it by IP, because name might be different
```

Comment:
Maybe enhance this comment to say "This can specifically happen if the IP was changed from ephemeral to static which results in a new name for the IP.